topology/probes/blockdev/blockdev.go

```diff
@@ -0,0 +1,462 @@
+	if err != nil {
+		p.Ctx.Logger.Error(err)
+		return nil
+	}
+	p.Groups[name] = groupNode
+
+	return groupNode
+}
+
+// addGroup adds a group to the graph
+func (p *ProbeHandler) addGroup(blockdev BlockDevice, WWN string) *graph.Node {
+	groupName := p.getID(blockdev)
+
+	return p.addGroupByName(groupName, WWN)
+}
+
+func (p *ProbeHandler) getMetaData(blockdev BlockDevice, childCount int, parentWWN string) (metadata graph.Metadata) {
+	var blockdevMetadata Metadata
+	var nodeType string
+
+	// The nodeType maps to the icon is used for display
+	if childCount > 0 {
+		if blockdev.Type == lvmType {
+			nodeType = lvmNodeType
+		} else {
+			nodeType = blockdevNodeType
+		}
+	} else {
+		nodeType = leafNodeType
+	}
+
+	// JSON for multipath devices doesn't include the WWN, so get it from
+	// the parent for a multipathType
+	var WWN string
+	if blockdev.Type == multipathType {
+		WWN = parentWWN
+	} else {
+		WWN = blockdev.WWN
+	}
+
+	blockdevMetadata = Metadata{
+		ID:   p.getID(blockdev),
+		Name: p.getName(blockdev),
+
+		Alignment:    blockdev.Alignment,
+		DiscAln:      blockdev.DiscAln,
+		DiscGran:     blockdev.DiscGran,
+		DiscMax:      blockdev.DiscMax,
+		DiscZero:     blockdev.DiscZero,
+		Fsavail:      blockdev.Fsavail,
+		Fssize:       blockdev.Fssize,
+		Fstype:       blockdev.Fstype,
+		FsusePercent: blockdev.FsusePercent,
+		Fsused:       blockdev.Fsused,
+		Group:        blockdev.Group,
+		Hctl:         blockdev.Hctl,
+		Hotplug:      blockdev.Hotplug,
+		Kname:        blockdev.Kname,
+		Label:        blockdev.Label,
+		LogSec:       blockdev.LogSec,
+		MajMin:       blockdev.MajMin,
+		MinIo:        blockdev.MinIo,
+		Mode:         blockdev.Mode,
+		Model:        blockdev.Model,
+		Mountpoint:   blockdev.Mountpoint,
+		OptIo:        blockdev.OptIo,
+		Owner:        blockdev.Owner,
+		Partflags:    blockdev.Partflags,
+		Partlabel:    blockdev.Partlabel,
+		Parttype:     blockdev.Parttype,
+		Partuuid:     blockdev.Partuuid,
+		Path:         blockdev.Path,
+		PhySec:       blockdev.PhySec,
+		Pttype:       blockdev.Pttype,
+		Ptuuid:       blockdev.Ptuuid,
+		Ra:           blockdev.Ra,
+		Rand:         blockdev.Rand,
+		Rev:          blockdev.Rev,
+		Rm:           blockdev.Rm,
+		Ro:           blockdev.Ro,
+		Rota:         blockdev.Rota,
+		RqSize:       blockdev.RqSize,
+		Sched:        blockdev.Sched,
+		Serial:       blockdev.Serial,
+		Size:         blockdev.Size,
+		State:        blockdev.State,
+		Subsystems:   blockdev.Subsystems,
+		Tran:         blockdev.Tran,
+		Type:         blockdev.Type,
+		UUID:         blockdev.UUID,
+		Vendor:       blockdev.Vendor,
+		Wsame:        blockdev.Wsame,
+		WWN:          WWN,
+	}
+	metadata = graph.Metadata{
+		"Path":       p.getPath(blockdev),
+		"Type":       nodeType,
+		"Name":       p.getName(blockdev),
+		"MajorMinor": blockdev.MajMin,
+		"WWN":        WWN,
+		"Manager":    managerType,
+		"BlockID":    p.getID(blockdev),
+		"MetaData":   blockdevMetadata,
```

Comment:
We are already in metadata, so a MetaData field is a bit confusing.
What about `BlockDevice` or `BlockDev` ? By the way, you should use the same as in the `init` function of analyzer/server.go

I would also suggest to move "BlockID", "WWN" and "MajorMinor" to the Metadata struct you added

Reply:
ok